rust/src/protocol/mod.rs

```diff
@@ -640,6 +640,111 @@ pub struct Protocol {
     /// Minimum version of the Delta write protocol a client must implement to correctly read the
     /// table.
     pub min_writer_version: i32,
+    /// Table features are missing from older versions
+    /// The table features this reader supports
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub reader_features: Option<HashSet<ReaderFeatures>>,
+    /// Table features are missing from older versions
+    /// The table features this writer supports
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub writer_features: Option<HashSet<WriterFeatures>>,
+}
+
+/// Features table readers can support as well as let users know
+/// what is supported
+#[derive(Serialize, Deserialize, Debug, Copy, Clone, Eq, PartialEq, Hash)]
+pub enum ReaderFeatures {
```

Comment:
I like the idea of having these features, but it does mean we error when we see a feature we don't recognize when parsing, right? There is a `serde::other` tag that I think we could use here. ([docs](https://serde.rs/variant-attrs.html)).

Reply:
Yeah, I like that lemme update for that